### PR TITLE
doc: Add doc for Bluetooth mesh SAR Config models

### PR DESCRIPTION
**Pull request by mia-ko**
Adding rst files describing SAR Configuration
models, SAR Configuration Client and SAR Configuration Server.

### CODE REVIEW

File: doc/connectivity/bluetooth/api/mesh/sar_cfg_srv.rst [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+* SAR Discard Timeout: This value defines the time since the last successfully received segment before giving up the reception of a segmented message.
+  Use the Kconfig option :kconfig:option:`CONFIG_BT_MESH_SAR_RX_DISCARD_TIMEOUT` to set the value.
+
+* SAR Acknowledgment Delay Increment: This value controls the delay increment of an interval used for delaying the transmission of an acknowledgment message after receiving a new segment.
+  Use the Kconfig option :kconfig:option:`CONFIG_BT_MESH_SAR_RX_ACK_DELAY_INC` to set a value.
+  The increment is measured in segments and calculated using the following formula::
+
+   :kconfig:option:`CONFIG_BT_MESH_SAR_RX_ACK_DELAY_INC` + 1.5
+
+* SAR Acknowledgment Retransmissions Count: This value defines the total number of retranmissions of an acknowledgment message that the stack will additionally send when the size of segments in a segmented message is above the :kconfig:option:`CONFIG_BT_MESH_SAR_RX_SEG_THRESHOLD` value.
```
Comment (akredalen) [inline doc/connectivity/bluetooth/api/mesh/sar_cfg_srv.rst:75]: This configuration is missing its :kconfig:option: (see comment above about first mention)

File: doc/connectivity/bluetooth/api/mesh/sar_cfg_srv.rst [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+* SAR Receiver Segment Interval Step: This value defines the segments reception interval step used for delaying the transmission of an acknowledgment message after receiving a new segmnet.
```
Comment (akredalen) [inline doc/connectivity/bluetooth/api/mesh/sar_cfg_srv.rst:77]: Same here.  :kconfig:option: only used in formula which is inconsistent with other definitions in the file

Comment (akredalen) [inline doc/connectivity/bluetooth/api/mesh/sar_cfg_srv.rst:77]: Also:
```suggestion
* SAR Receiver Segment Interval Step: This value defines the segments reception interval step used for delaying the transmission of an acknowledgment message after receiving a new segment.
```

File: doc/connectivity/bluetooth/api/mesh/sar_cfg.rst [hunk 3]
```diff
@@ -0,0 +1,372 @@
...
+
+If the Upper Transport PDU fits, it is sent in a single Lower Transport PDU.
+An Upper Transport PDU that fits into one Lower Transport PDU can also be sent as a single-segment segmented message when acknowledgment
+by the lower transport layer is required.
+
+If the Upper Transport PDU doesn’t fit, the lower transport layer divides the Upper Transport PDU into two or more Lower Transport PDUs.
+If a segmented Upper Transport PDU is tagged with additional metadata, each Lower Transport PDU of the segmented Upper Transport PDU is
+tagged with the same metadata.
+
+Delivery of a segmented message is acknowledged by the lower transport layer of the receiving node.
```
Comment (akredalen) [inline doc/connectivity/bluetooth/api/mesh/sar_cfg.rst:44]:
```suggestion
Delivery of a segmented message is acknowledged by the lower transport layer of the receiving node,  while an unsegmented message delivery isn't acknowledged.
```

File: doc/connectivity/bluetooth/api/mesh/sar_cfg.rst [hunk 4]
```diff
@@ -0,0 +1,372 @@
...
+Delivery of an unsegmented message is not acknowledged.
```
Comment (akredalen) [inline doc/connectivity/bluetooth/api/mesh/sar_cfg.rst:45]:
```suggestion
```

**Review by akredalen — CHANGES_REQUESTED** [review]
Nice work! Just a few comments / minor edits

File: doc/connectivity/bluetooth/api/mesh/sar_cfg.rst [hunk 5]
```diff
@@ -0,0 +1,372 @@
+.. _bluetooth_mesh_sar_cfg:
+
+Segmentation and reassembly (SAR)
+#################################
+
+Segmentation and reassembly is a mechanism used by the lower transport layer.
+It provides a way of handling larger upper transport layer messages in a mesh network, with a purpose of enhancing the Bluetooth mesh throughput.
+
+The lower transport layer takes PDUs from the upper transport layer and sends them to the lower transport layer on a peer device.
```
Comment (Balaklaka) [inline doc/connectivity/bluetooth/api/mesh/sar_cfg.rst:9]:
```suggestion
The lower transport layer takes Upper Transport PDUs from the upper transport layer and sends them to the lower transport layer on a peer device.
```

File: doc/connectivity/bluetooth/api/mesh/sar_cfg.rst [hunk 6]
```diff
@@ -0,0 +1,372 @@
...
+For longer packets, which do not fit into a single Lower Transport PDU, the lower transport layer performs segmentation, splitting the PDU
+into multiple Lower Transport PDUs.
```
Comment (Balaklaka) [inline doc/connectivity/bluetooth/api/mesh/sar_cfg.rst:11]:
```suggestion
For longer packets, which do not fit into a single Lower Transport PDU, the lower transport layer performs segmentation, splitting the Upper Transport PDU
into multiple segments.
```

File: doc/connectivity/bluetooth/api/mesh/sar_cfg.rst [hunk 7]
```diff
@@ -0,0 +1,372 @@
...
+The receiving lower transport layer on the other device reassembles the segments into a single upper transport layer PDU before passing it
```
Comment (Balaklaka) [inline doc/connectivity/bluetooth/api/mesh/sar_cfg.rst:12]:
```suggestion
The receiving lower transport layer on the other device reassembles the segments into a single Upper Transport PDU before passing it
```

File: doc/connectivity/bluetooth/api/mesh/sar_cfg.rst [hunk 8]
```diff
@@ -0,0 +1,372 @@
...
+Never more than one segmented message is transmitted to the same destination.
+The segments are sent in order of ``SegO`` field value, starting with the lowest value.
+Transmission of segments is separated by a segment transmission interval (value indicated by the SAR Segment Interval Step state).
+
+It is not possible to send a segmented message when the ``SEQ`` field value is 8192 greater than the ``SeqAuth`` value.
+If a segmented message has not been acknowledged by the time that the ``SEQ`` field gets this value, the transmission of the Upper
+Transport PDU is canceled and the Lower Transport PDUs are delivered to the network layer.
+
+Segments can be sent to a unicast address or to a group or a virtual address.
+
```
Comment (Balaklaka) [inline doc/connectivity/bluetooth/api/mesh/sar_cfg.rst:85]: Is it possible to merge the "Transmitting to a ...." into one and just explain what it unique for unicast address, as there is a lot of repeating text. It makes it less to read and maintain. Same for retransmition.

**Review by Balaklaka — CHANGES_REQUESTED** [review]

File: doc/connectivity/bluetooth/api/mesh/sar_cfg.rst [hunk 9]
```diff
@@ -0,0 +1,238 @@
...
+* SAR Receiver Segment Interval Step
+
+SAR Segment Interval Step
+=========================
+
+SAR Segment Interval Step state holds a value that controls the interval between transmissions of segments of a segmented message.
+Use the :kconfig:option:`CONFIG_BT_MESH_SAR_TX_SEG_INT_STEP` Kconfig option to set a value.
+Segment transmission interval is then calculated using the following formula::
+
+(:kconfig:option:`CONFIG_BT_MESH_SAR_TX_SEG_INT_STEP` + 1) * 10 ms
```
Comment (kartben) [inline doc/connectivity/bluetooth/api/mesh/sar_cfg.rst:131]: you can't use the `:kconfig:option:` role within a code block as it doesn't render

image

File: doc/connectivity/bluetooth/api/mesh/sar_cfg.rst [hunk 10]
```diff
@@ -0,0 +1,238 @@
...
+The initial value of the SAR Multicast Retransmissions timer is set to the multicast retransmissions interval.
+
+When the lower transport layer receives a segment of a segmented message, it starts an SAR Discard timer.
+The discard timer tells how long the lower transport layer waits before discarding the segmented message the segment belongs to.
+The initial value of the SAR Discard timer is the discard timeout value indicated by the `SAR Discard Timeout`_ state.
+
+SAR Acknowledgment timer holds the time before a Segment Acknowledgment message is sent for a received segment.
+The initial value of the SAR Acknowledgment timer is calculated using the following formula::
+
+[min(SegN + 0.5 , acknowledgment delay increment) * segment reception interval]
```
Comment (kartben) [inline doc/connectivity/bluetooth/api/mesh/sar_cfg.rst:83]: You should consider using actual math formulas as they render much more nicely.

```suggestion
The initial value of the SAR Acknowledgment timer is calculated using the following formula:

.. math::

   \min(\text{SegN} + 0.5, \text{acknowledgment delay increment}) \times \text{segment reception interval}
```

Alternatively, this simplified form works too

```suggestion
The initial value of the SAR Acknowledgment timer is calculated using the following formula:

.. math::

   min(SegN + 0.5 , acknowledgment~delay~increment) * segment~reception~interval
```

Finally, it's also possible to use the `:math:` role if you want the formula in-line with the text (``blah blah :math:`my formula` blah blah``).

File: doc/connectivity/bluetooth/api/mesh/sar_cfg.rst [hunk 11]
```diff
@@ -0,0 +1,238 @@
...
+Segmentation and reassembly (SAR)
+#################################
+
+Segmentation and reassembly provides a way of handling larger upper transport layer messages in a
+mesh network, with a purpose of enhancing the Bluetooth mesh throughput. The segmentation and
+reassembly mechanism is used by the lower transport layer.
+
+The lower transport layer takes Upper Transport PDUs from the upper transport layer and sends them to the lower transport layer on a peer device.
+If the Upper Transport PDU fits, it is sent in a single Lower Transport PDU.
+For longer packets, which do not fit into a single Lower Transport PDU, the lower transport layer performs segmentation, splitting the Upper Trasport
```
Comment (kartben) [inline doc/connectivity/bluetooth/api/mesh/sar_cfg.rst:12]:
```suggestion
For longer packets, which do not fit into a single Lower Transport PDU, the lower transport layer performs segmentation, splitting the Upper Transport
```

**Review by kartben — DISMISSED** [review]

File: doc/connectivity/bluetooth/api/mesh/sar_cfg.rst [hunk 12]
```diff
@@ -0,0 +1,238 @@
...
+The initial value of the SAR Discard timer is the discard timeout value indicated by the `SAR Discard Timeout`_ state.
+
+SAR Acknowledgment timer holds the time before a Segment Acknowledgment message is sent for a received segment.
+The initial value of the SAR Acknowledgment timer is calculated using the following formula::
+
+[min(SegN + 0.5 , acknowledgment delay increment) * segment reception interval]
+
+Four counters are related to SAR behavior:
+
+* Unicast retransmissions count (see `SAR Unicast Retransmissions Count`_ state and `SAR Unicast Retransmissions Without Progress Count`_ state)
```
Comment (Balaklaka) [inline doc/connectivity/bluetooth/api/mesh/sar_cfg.rst:92]:
```suggestion
* Two Unicast retransmissions counts (see `SAR Unicast Retransmissions Count`_ state and `SAR Unicast Retransmissions Without Progress Count`_ state)
```

File: doc/connectivity/bluetooth/api/mesh/sar_cfg.rst [hunk 13]
```diff
@@ -0,0 +1,243 @@
...
+Use the Kconfig option :kconfig:option:`CONFIG_BT_MESH_SAR_TX_MULTICAST_RETRANS_INT` to set a value that is used to calculate the interval
+using the following formula::
+
+(CONFIG_BT_MESH_SAR_TX_MULTICAST_RETRANS_INT + 1) * 25 ms
+
+SAR Discard Timeout
+===================
+
+The value of this state defines the time since the last successfully received segment before giving up the reception of a segmented message.
+Use the Kconfig option :kconfig:option:`CONFIG_BT_MESH_SAR_RX_DISCARD_TIMEOUT` to set the value for this state.
```
Comment (Balaklaka) [inline doc/connectivity/bluetooth/api/mesh/sar_cfg.rst:194]: What unit is this given in? (seconds/milliseconds)

**Review by Balaklaka — CHANGES_REQUESTED** [review]

File: doc/connectivity/bluetooth/api/mesh/sar_cfg.rst [hunk 14]
```diff
@@ -0,0 +1,243 @@
...
+Set the ``send rel`` flag (see :c:struct:`bt_mesh_msg_ctx`) to use the reliable message transmission and acknowledge single-segment segmented messages.
+
+The transport layer is able to transport up to 32 segments with its SAR mechanism, with a maximum message (PDU) size of 384 octets.
+To configure number of segments and the message size for the Bluetooth mesh stack, use the following Kconfig options:
+
+* :kconfig:option:`CONFIG_BT_MESH_SEG_BUFS` to set the number of segment buffers.
+* :kconfig:option:`CONFIG_BT_MESH_RX_SEG_MAX` to set the maximum number of segments in an incoming message.
+* :kconfig:option:`CONFIG_BT_MESH_TX_SEG_MAX` to set the maximum number of segments in an outcoming message.
+
+Never more than one segmented message is transmitted at the time, but the Bluetooth mesh stack allows you to define buffers that will queue a defined
```
Comment (akredalen) [inline doc/connectivity/bluetooth/api/mesh/sar_cfg.rst:30]: Maybe something like this instead: "Only one segmented message is transmitted at a time, but the Bluetooth mesh stack allows you to **configure** buffers that can queue a specified number of incoming or outgoing segmented messages."

File: doc/connectivity/bluetooth/api/mesh/sar_cfg.rst [hunk 15]
```diff
@@ -0,0 +1,243 @@
...
+number of ingoing or outgoing segmented messages.
+Use Kconfig options :kconfig:option:`CONFIG_BT_MESH_TX_SEG_MSG_COUNT` and :kconfig:option:`CONFIG_BT_MESH_RX_SEG_MSG_COUNT` to set the maximum number
+of queued outgoing and incoming segmented messages respectively.
+
+Incoming and outgoing segmented messages share the same segment buffer, which means that they allocate their segments from the same pool.
+Both incoming and outgoing messages allocate segments at the start of the transaction, but release them at the different time.
```
Comment (akredalen) [inline doc/connectivity/bluetooth/api/mesh/sar_cfg.rst:36]: I am a bit confused by this sentence. "but release them at the different time", what is meant by this? Should it be "at different times"?

File: doc/connectivity/bluetooth/api/mesh/sar_cfg.rst [hunk 16]
```diff
@@ -0,0 +1,243 @@
...
+   :maxdepth: 1
+
+   sar_cfg_cli
+   sar_cfg_srv
+
+Intervals, timers and counters
+******************************
+
+Transmission of segments is separated by a segment transmission interval (see the `SAR Segment Interval Step`_ state).
+Other time intervals and delays available for the segmentation and reassembly are:
```
Comment (akredalen) [inline doc/connectivity/bluetooth/api/mesh/sar_cfg.rst:59]:
```suggestion
Other configurable time intervals and delays available for the segmentation and reassembly are:
```

File: doc/connectivity/bluetooth/api/mesh/sar_cfg.rst [hunk 17]
```diff
@@ -0,0 +1,243 @@
...
+The initial value of the SAR Unicast Retransmissions timer depends on the value of the TTL field of the message.
+If the TTL field value is greater than ``0``, the initial value for the timer is set according to the following formula::
+
+[unicast retransmissions interval step + unicast retransmissions interval increment * (TTL - 1)]
+
+If the TTL field value is ``0``, the initial value of the timer is set to the unicast retransmissions interval step.
+
+The initial value of the SAR Multicast Retransmissions timer is set to the multicast retransmissions interval.
+
+When the lower transport layer receives a segment of a segmented message, it starts an SAR Discard timer.
```
Comment (akredalen) [inline doc/connectivity/bluetooth/api/mesh/sar_cfg.rst:76]:
```suggestion
When the lower transport layer receives a message segment, it starts a SAR Discard timer.
```

**Review by akredalen — CHANGES_REQUESTED** [review]

File: doc/connectivity/bluetooth/api/mesh/sar_cfg_cli.rst [hunk 18]
```diff
@@ -0,0 +1,34 @@
...
+The SAR Receiver procedure is used to determine and configure the SAR Receive state of a SAR Configuration Server.
+Function calls :c:func:`bt_mesh_sar_cfg_cli_receiver_get` and :c:func:`bt_mesh_sar_cfg_cli_receiver_set` are used to get and set the SAR Receiver state of the
+Target node respectively.
+
+For more information about the two states, see :ref:`bt_mesh_sar_cfg_states`.
+
+An element can send any SAR Configuration Client message at any time to query or change the states supported by the SAR Configuration Server model of a peer node.
+The SAR Configuration Client model only accepts messages encrypted with the device key of the node supporting the SAR Configuration Server model.
+
+If present, the SAR Configuration Client model must only be instantiated on the primary element.
```
[the remark this answers is not in the document]
Comment (akredalen) [reply doc/connectivity/bluetooth/api/mesh/sar_cfg_cli.rst:27]: I would like to keep the "only" here. So "shall only" then?

File: doc/connectivity/bluetooth/api/mesh/sar_cfg_srv.rst [hunk 19]
```diff
@@ -0,0 +1,29 @@
...
+
+The SAR Configuration Server model defines two states, SAR Transmitter state and SAR Receiver state.
+For more information about the two states, see :ref:`bt_mesh_sar_cfg_states`.
+
+The model also supports the SAR Transmitter and SAR Receiver get and set messages.
+
+The SAR Configuration Server model does not have an API of its own, but relies on a :ref:`bluetooth_mesh_sar_cfg_cli` to control it.
+The SAR Configuration Server model only accepts messages encrypted with the node’s device key.
+
+If present, the model must be instantiated on the primary element.
```
[the remark this answers is not in the document]
Comment (akredalen) [reply doc/connectivity/bluetooth/api/mesh/sar_cfg_srv.rst:22]: See the comment above for the client.

File: doc/connectivity/bluetooth/api/mesh/sar_cfg.rst (hunk 17 above)
Comment (mia-ko) [reply doc/connectivity/bluetooth/api/mesh/sar_cfg.rst:76]: It depends on whether you pronounce SAR as [sɑr] or [es/eɪ/ar/]. Not sure which one should be used.

Comment (akredalen) [reply doc/connectivity/bluetooth/api/mesh/sar_cfg.rst:76]: I believe [sar] is the most common but this is just nitpicking :)

File: doc/connectivity/bluetooth/api/mesh/sar_cfg.rst (hunk 15 above)
Comment (mia-ko) [reply doc/connectivity/bluetooth/api/mesh/sar_cfg.rst:36]: at different times, for sure :)

File: doc/connectivity/bluetooth/api/mesh/sar_cfg.rst [hunk 20]
```diff
@@ -0,0 +1,238 @@
...
+To configure number of segments and the message size for the Bluetooth mesh stack, use the following Kconfig options:
+
+* :kconfig:option:`CONFIG_BT_MESH_SEG_BUFS` to set the number of segment buffers.
+* :kconfig:option:`CONFIG_BT_MESH_RX_SEG_MAX` to set the maximum number of segments in an incoming message.
+* :kconfig:option:`CONFIG_BT_MESH_TX_SEG_MAX` to set the maximum number of segments in an outcoming message.
+
+Never more than one segmented message is transmitted at the time, but the Bluetooth mesh stack allows you to define buffers that will queue a defined
+number of ingoing or outgoing segmented messages.
+Use Kconfig options :kconfig:option:`CONFIG_BT_MESH_TX_SEG_MSG_COUNT` and :kconfig:option:`CONFIG_BT_MESH_RX_SEG_MSG_COUNT` to set the maximum number
+of queued outgoing and incoming segmented messages respectively.
```
[the remark this answers is not in the document]
Comment (mia-ko) [reply doc/connectivity/bluetooth/api/mesh/sar_cfg.rst:33]: :kconfig:option:`CONFIG_BT_MESH_TX_SEG_MSG_COUNT` defines outgoing messages, right? And that one is mentioned first in the sentence. That's why outgoing first, and then incoming. Would you like me to swap the Kconfig options to make your suggestion right?

File: doc/connectivity/bluetooth/api/mesh/sar_cfg.rst [hunk 21]
```diff
@@ -0,0 +1,238 @@
...
+
+The transport layer is able to transport up to 32 segments with its SAR mechanism, with a maximum message (PDU) size of 384 octets.
+To configure number of segments and the message size for the Bluetooth mesh stack, use the following Kconfig options:
+
+* :kconfig:option:`CONFIG_BT_MESH_SEG_BUFS` to set the number of segment buffers.
+* :kconfig:option:`CONFIG_BT_MESH_RX_SEG_MAX` to set the maximum number of segments in an incoming message.
+* :kconfig:option:`CONFIG_BT_MESH_TX_SEG_MAX` to set the maximum number of segments in an outcoming message.
+
+Never more than one segmented message is transmitted at the time, but the Bluetooth mesh stack allows you to define buffers that will queue a defined
+number of ingoing or outgoing segmented messages.
```
[the remark this answers is not in the document]
Comment (mia-ko) [reply doc/connectivity/bluetooth/api/mesh/sar_cfg.rst:31]: Right...
It's been a few ingoing and outcoming messages LOL Unbelievable!

File: doc/connectivity/bluetooth/api/mesh/sar_cfg_cli.rst (hunk 18 above)
[the remark this answers is not in the document]
Comment (mia-ko) [reply doc/connectivity/bluetooth/api/mesh/sar_cfg_cli.rst:27]: I have checked docs for our other foundation models and I see that we've used
- must be 
- should be
- and can be
Most of them using must.
I wouldn't like to introduce yet another way of describing it. Are we ok with continuing with must only?

File: doc/connectivity/bluetooth/api/mesh/sar_cfg.rst (hunk 13 above)
Comment (mia-ko) [reply doc/connectivity/bluetooth/api/mesh/sar_cfg.rst]: seconds

File: doc/connectivity/bluetooth/api/mesh/sar_cfg.rst (hunk 10 above)
Comment (kartben) [reply doc/connectivity/bluetooth/api/mesh/sar_cfg.rst]: FWIW I would recommend you use the math directive everywhere, not just in this particular instance.
Rendered doc available at https://builds.zephyrproject.io/zephyr/pr/61886/docs/connectivity/bluetooth/api/mesh/sar_cfg.html , in case you are not aware and/or are not testing locally.

File: doc/connectivity/bluetooth/api/mesh/sar_cfg.rst [hunk 22]
```diff
@@ -0,0 +1,247 @@
...
+==================================
+
+The SAR Receiver Segment Interval Step defines the segments reception interval step used for delaying the transmission of an acknowledgment message after receiving a new segment.
+The interval is measured in milliseconds.
+
+Use the Kconfig option :kconfig:option:`CONFIG_BT_MESH_SAR_RX_SEG_INT_STEP` to set a value that will be used to calculate the interval using the following formula::
+
+(CONFIG_BT_MESH_SAR_RX_SEG_INT_STEP + 1) * 10 ms
+
+API reference
```
Comment (PavelVPV) [inline doc/connectivity/bluetooth/api/mesh/sar_cfg.rst:240]: This can be removed. This API is not public.

File: doc/connectivity/bluetooth/api/mesh/sar_cfg_cli.rst [hunk 23]
```diff
@@ -0,0 +1,34 @@
+.. _bluetooth_mesh_sar_cfg_cli:
+
+SAR Configuration Client
+########################
+
+The SAR Configuration Client model is a foundation model defined by the Bluetooth mesh specification.
+It is an optional model, enabled with the :kconfig:option:`CONFIG_BT_MESH_SAR_CFG_CLI` configuration option.
+
+The SAR Configuration Client model is introduced in the `Bluetooth Mesh Protocol Specification version 1.1 <https://www.bluetooth.com/specifications/mesh-specifications/>`_,
```
Comment (PavelVPV) [inline doc/connectivity/bluetooth/api/mesh/sar_cfg_cli.rst:9]: Same comment as above. I'm not sure the link is needed in every file.

**Review by akredalen — CHANGES_REQUESTED** [review]

**Review by PavelVPV — CHANGES_REQUESTED** [review]

File: doc/connectivity/bluetooth/api/mesh/sar_cfg.rst (hunk 9 above)
Comment (mia-ko) [reply doc/connectivity/bluetooth/api/mesh/sar_cfg.rst]: Of course, I forgot to think of that. Removed the :kconfig:option:``

File: doc/connectivity/bluetooth/api/mesh/sar_cfg.rst (hunk 10 above)
Comment (mia-ko) [reply doc/connectivity/bluetooth/api/mesh/sar_cfg.rst]: Thanks Benjamin :)
Yeah, I know how they render. Personally I prefer code block rendition over the one using math, it's just something stylistically off with that one for me, for some reason I cannot quite explain. That's why I've been holding back.
But, I'll give it a go and see if it starts growing on me LOL

File: doc/connectivity/bluetooth/api/mesh/sar_cfg.rst [hunk 24]
```diff
@@ -0,0 +1,243 @@
...
+Incoming and outgoing segmented messages share the same segment buffer, which means that they allocate their segments from the same pool.
+Both incoming and outgoing messages allocate segments at the start of the transaction, but release them at the different time.
+While the outgoing segmented message releases its segments one by one as soon as they are acknowledged by the receiver, the incoming message will not
+release the segments until the message is fully received.
+Keep this in mind when defining the size of the buffers.
+
+SAR does not impose extra overhead on the access layer payload per segment.
+
+Segmentation and reassembly (SAR) Configuration models are foundation models defined by the
+`Bluetooth Mesh Protocol Specification version 1.1 <https://www.bluetooth.com/specifications/mesh-specifications/>`_.
```
[the remark this answers is not in the document]
Comment (mia-ko) [reply doc/connectivity/bluetooth/api/mesh/sar_cfg.rst:44]: This is the first and only place the spec is referenced on this page.
I am not sure what page you mean, maybe mesh.rst?
I also see we are only mentioning v1.0.1 there, no 1.1.

[the remark this answers is not in the document]
Comment (PavelVPV) [reply doc/connectivity/bluetooth/api/mesh/sar_cfg.rst:44]: > This is the first and only place the spec is referenced on this page.
I am not sure what page you mean, maybe mesh.rst?

Yes, I meant mesh.rst. Otherwise, we need to add the reference to every model's doc page which is too much I think. 

> I also see we are only mentioning v1.0.1 there, no 1.1.

This should be fixed (not necessarily in your PR, but up to you).

File: doc/connectivity/bluetooth/api/mesh/sar_cfg_cli.rst (hunk 23 above)
Comment (mia-ko) [reply doc/connectivity/bluetooth/api/mesh/sar_cfg_cli.rst:9]: I'm thinking having the link once, the first time it is introduced on a page, cannot hurt.
At least if looking from the perspective of our new doc portal, where the user will search for specific topics rather than read through page by page in a traditional manner. Having the link available on the page that user reads, instead of having to go back to a page where the link was introduced is better. We are incouraged to have an "every page is a first page" philosophy.
But I can remove it if you feel strongly about it.
Remove or stay?

Comment (PavelVPV) [reply doc/connectivity/bluetooth/api/mesh/sar_cfg_cli.rst:9]: What I **really** don't like is different styles in the same code base or documentation. What I expect will happen next is that here we will keep the link, but other pages will stay like they are right now (without the link). If we want to add link here, then we have to be consistent and add it to other pages as well (I don't see anything special in SAR models). 

I raised the concern of not having a common pattern/style for upstream docs on the last retrospective and created a follow up task NCSDK-23389. This particular case could be one of the points in that task.

P.S. This link doesn't really exist anymore, it redirects to all Bluetooth specs page.

**Review by akredalen — DISMISSED** [review]

File: doc/connectivity/bluetooth/api/mesh/sar_cfg.rst [hunk 25]
```diff
@@ -0,0 +1,276 @@
...
+
+When sending a segmented message to a unicast address, the unacknowledged segments are repeated
+the :kconfig:option:`CONFIG_BT_MESH_TX_SEG_RETRANS_COUNT` number of times before the transmission
+is considered as failed. The same option configures a number of retransmissions to a group or virtual
+address, but the transmission always succeedes after retransmitting all segments the configured
+number of times.
+
+The timeout between each retransmission to a unicast address and a group or virtual address is
+configured by Kconfig options :kconfig:option:`CONFIG_BT_MESH_TX_SEG_RETRANS_TIMEOUT_UNICAST` and
+:kconfig:option:`CONFIG_BT_MESH_TX_SEG_RETRANS_TIMEOUT_GROUP` respectively.
```
Comment (Balaklaka) [inline doc/connectivity/bluetooth/api/mesh/sar_cfg.rst:55]: This was a little bit hard to follow, "unicast address and a group or virtual address". First there is "and" then "or"? Which kconfig option is used for virtual address?

File: doc/connectivity/bluetooth/api/mesh/sar_cfg_srv.rst [hunk 26]
```diff
@@ -0,0 +1,29 @@
...
+
+The SAR Configuration Server model defines two states, SAR Transmitter state and SAR Receiver state.
+For more information about the two states, see :ref:`bt_mesh_sar_cfg_states`.
+
+The model also supports the SAR Transmitter and SAR Receiver get and set messages.
+
+The SAR Configuration Server model does not have an API of its own, but relies on a :ref:`bluetooth_mesh_sar_cfg_cli` to control it.
+The SAR Configuration Server model only accepts messages encrypted with the node’s device key.
+
+If present, the model must only be instantiated on the primary element.
```
[the remark this answers is not in the document]
Comment (akredalen) [reply doc/connectivity/bluetooth/api/mesh/sar_cfg_srv.rst]: I suggested adding the word "only" to emphasize that the model cannot be instantiated on any other element. Without the word "only," it might be interpreted as allowing you to also adding it to the next elements as long as the model is present on the primary element.

The spec says: "If supported, the model **shall** be supported by a primary element **and** shall not be supported by any secondary elements."

**Review by Balaklaka — APPROVED** [review]
I am approving this, as my only comment is about this "must vs shall" and "only" wording.

File: doc/connectivity/bluetooth/api/mesh.rst [hunk 27]
```diff
@@ -5,7 +5,10 @@ Bluetooth Mesh Profile
 
 The Bluetooth mesh profile adds secure wireless multi-hop communication for
 Bluetooth Low Energy. This module implements the
-`Bluetooth Mesh Profile Specification v1.0.1 <https://www.bluetooth.com/specifications/mesh-specifications/>`_.
+`Bluetooth Mesh Profile Specification v1.0.1 <https://www.bluetooth.com/specifications/specs/mesh-profile-1-0-1/>`_.
+
+Implementation of the draft `Bluetooth Mesh Protocol Specification v1.1 <https://www.bluetooth.com/specifications/specs/mesh-protocol-1-1/>`_
```
Comment (PavelVPV) [inline doc/connectivity/bluetooth/api/mesh.rst:10]: Something with rendering here

File: doc/connectivity/bluetooth/api/mesh/sar_cfg.rst [hunk 28]
```diff
@@ -0,0 +1,278 @@
...
+* Segment reception interval (see the `SAR Receiver Segment Interval Step`_ state).
+* Acknowledgment delay increment (see the `SAR Acknowledgment Delay Increment`_ state).
+
+When the last segment marked as unacknowledged is transmitted, the lower transport layer starts a retransmissions timer.
+The initial value of the SAR Unicast Retransmissions timer depends on the value of the TTL field of the message.
+If the TTL field value is greater than ``0``, the initial value for the timer is set according to the following formula:
+
+.. math::
+
+   [unicast~retransmissions~interval~step + unicast~retransmissions~interval~increment \times (TTL - 1)]
```
Comment (PavelVPV) [inline doc/connectivity/bluetooth/api/mesh/sar_cfg.rst:93]: Minor: do we need square brackets here?

**Review by akredalen — DISMISSED** [review]

**Comment by mia-ko** [comment]
> I am approving this, as my only comment is about this "must vs shall" and "only" wording.

I think we can keep 'must'. It is not wrong, and within Nordic we have this rule of avoiding shall/should and using must/will instead. And even though this is not NCS documentation, I am still quite influenced by this rule.
When it comes to 'only', I agree with Stine on this one. Stating that something must exist in one place, does not explicitly mean that it cannot exist elsewhere in addition. Using only, we make sure that it is on primary only. If that is ok with you, we do no changes.

File: doc/connectivity/bluetooth/api/mesh.rst (hunk 27 above)
Comment (PavelVPV) [reply doc/connectivity/bluetooth/api/mesh.rst:10]: Sorry, now it looks fine. In one of the previous builds it didn't render the link correctly.

File: doc/connectivity/bluetooth/api/mesh/sar_cfg.rst (hunk 28 above)
Comment (PavelVPV) [reply doc/connectivity/bluetooth/api/mesh/sar_cfg.rst:93]: Yes

**Comment by PavelVPV** [comment]
> > I am approving this, as my only comment is about this "must vs shall" and "only" wording.
> 
> I think we can keep 'must'. It is not wrong, and within Nordic we have this rule of avoiding shall/should and using must/will instead. And even though this is not NCS documentation, I am still quite influenced by this rule. When it comes to 'only', I agree with Stine on this one. Stating that something must exist in one place, does not explicitly mean that it cannot exist elsewhere in addition. Using only, we make sure that it is on primary only. If that is ok with you, we do no changes.

We need to align other models to this rule as some don't have "only", some has "can be" instead of "must"...

**Review by PavelVPV — APPROVED** [review]

**Comment by PavelVPV** [comment]
> > > I am approving this, as my only comment is about this "must vs shall" and "only" wording.
> > 
> > 
> > I think we can keep 'must'. It is not wrong, and within Nordic we have this rule of avoiding shall/should and using must/will instead. And even though this is not NCS documentation, I am still quite influenced by this rule. When it comes to 'only', I agree with Stine on this one. Stating that something must exist in one place, does not explicitly mean that it cannot exist elsewhere in addition. Using only, we make sure that it is on primary only. If that is ok with you, we do no changes.
> 
> We need to align other models to this rule as some don't have "only", some has "can be" instead of "must"...

I'll fix this separately.

**Comment by akredalen** [comment]
> > We need to align other models to this rule as some don't have "only", some has "can be" instead of "must"...
> 
> I'll fix this separately.

I think @mia-ko is on this already and wanted to add an additional commit to this PR?

**Comment by PavelVPV** [comment]
> > > We need to align other models to this rule as some don't have "only", some has "can be" instead of "must"...
> > 
> > 
> > I'll fix this separately.
> 
> I think @mia-ko is on this already and wanted to add an additional commit to this PR?

Sorry, @mia-ko , I created it already: https://github.com/zephyrproject-rtos/zephyr/pull/62523

**Review by akredalen — APPROVED** [review]

**Comment by mia-ko** [comment]
> > > > We need to align other models to this rule as some don't have "only", some has "can be" instead of "must"...
> > > 
> > > 
> > > I'll fix this separately.
> > 
> > 
> > I think @mia-ko is on this already and wanted to add an additional commit to this PR?
> 
> Sorry, @mia-ko , I created it already: #62523

It's ok, as long as it gets done :) Thanks. I'll revert my changes.
